lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -322,7 +327,7 @@ struct ExpressionVisitor
 
     info.stream << ")\n";
 
-    if (weight != 0)
+    if (weight != 0 || !patterns.empty())
       info.stream << "( ! ";
```

Comment:
Currently, this `( !` is printed with out nice indentation:
```
(assert (let ((tmp_1 (exists ((tmp_2 Int) (tmp_3 Int))
( ! (let ((tmp_4 (= tmp_2 tmp_3)))
                             tmp_4) :weight 2))))
```

We could move this to just before the next `printExpression` call below to use the indentation added just before that call.